<a href="https://colab.research.google.com/github/mpaydar/381_GroupProject/blob/main/381_groupProject.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Overview
Data description “e-shop clothing 2008”

Variables:

1. YEAR (2008)

========================================================

2. MONTH -> from April (4) to August (8)

========================================================

3. DAY -> day number of the month

========================================================

4. ORDER -> sequence of clicks during one session

========================================================

5. COUNTRY -> variable indicating the country of origin of the IP address with the 
following categories:

1-Australia
2-Austria
3-Belgium
4-British Virgin Islands
5-Cayman Islands
6-Christmas Island
7-Croatia
8-Cyprus
9-Czech Republic
10-Denmark
11-Estonia
12-unidentified
13-Faroe Islands
14-Finland
15-France
16-Germany
17-Greece
18-Hungary
19-Iceland
20-India
21-Ireland
22-Italy
23-Latvia
24-Lithuania
25-Luxembourg
26-Mexico
27-Netherlands
28-Norway
29-Poland
30-Portugal
31-Romania
32-Russia
33-San Marino
34-Slovakia
35-Slovenia
36-Spain
37-Sweden
38-Switzerland
39-Ukraine
40-United Arab Emirates
41-United Kingdom
42-USA
43-biz (*.biz)
44-com (*.com)
45-int (*.int)
46-net (*.net)
47-org (*.org)

========================================================

6 . SESSION ID -> variable indicating session id (short record)

========================================================

7. PAGE 1 (MAIN CATEGORY) -> concerns the main product category:
1-trousers
2-skirts
3-blouses
4-sale


========================================================

8. COLOUR -> colour of product

1-beige
2-black
3-blue
4-brown
5-burgundy
6-gray
7-green
8-navy blue
9-of many colors
10-olive
11-pink
12-red
13-violet
14-white





========================================================

9 . PRICE -> price in US dollars

========================================================

10 . PRICE 2 -> variable informing whether the price of a particular product is higher than 
the average price for the entire product category

1-yes
2-no

========================================================

In [22]:
import pandas as pd


dataframe=pd.read_csv("https://raw.githubusercontent.com/mpaydar/381_GroupProject/main/GroupProject/e_shopDataset.csv")
dataframe.rename(columns={'page 1 (main category)': 'Product Category'},inplace=True)      # updating some of the attribute name to a better one 
dataframe=dataframe.drop(["page 2 (clothing model)","model photography","location","page"],axis=1)    # removing some of the attributes we don't need in our analysis 
dataframe.head()

year  month  day  order  country  session ID  Product Category  colour  \
0  2008      4    1      1       29           1                 1       1   
1  2008      4    1      2       29           1                 1       1   
2  2008      4    1      3       29           1                 2      10   
3  2008      4    1      4       29           1                 2       6   
4  2008      4    1      5       29           1                 2       4   

   price  price 2  
0     28        2  
1     33        2  
2     52        1  
3     38        2  
4     52        1

In [23]:
a=dataframe['year'].unique  
dataframe.nunique()

year                    1
month                   5
day                    31
order                 195
country                47
session ID          24026
Product Category        4
colour                 14
price                  20
price 2                 2
dtype: int64

All the data seems to be from 2008 , so there is really no use of using year as it's not giving us any information. However, from the nunique output we could see that there are several distinct days and months which could give us an insight as we dig more down into data.

In [24]:
dataframe=dataframe.drop('year',axis=1)
dataframe.head()

month  day  order  country  session ID  Product Category  colour  price  \
0      4    1      1       29           1                 1       1     28   
1      4    1      2       29           1                 1       1     33   
2      4    1      3       29           1                 2      10     52   
3      4    1      4       29           1                 2       6     38   
4      4    1      5       29           1                 2       4     52   

   price 2  
0        2  
1        2  
2        1  
3        2  
4        1

# Corrlation of attributes

In [25]:
dataframe.corr()

month       day     order   country  session ID  \
month             1.000000 -0.075098  0.026093  0.065502    0.970961   
day              -0.075098  1.000000 -0.021744  0.011551    0.161369   
order             0.026093 -0.021744  1.000000 -0.064709    0.021829   
country           0.065502  0.011551 -0.064709  1.000000    0.065869   
session ID        0.970961  0.161369  0.021829  0.065869    1.000000   
Product Category  0.017557 -0.001251  0.151693  0.132817    0.016123   
colour           -0.035279  0.005313  0.046427  0.009789   -0.032034   
price             0.012700 -0.002818 -0.064779 -0.045345    0.011669   
price 2          -0.014658  0.004254  0.000772 -0.017536   -0.012742   

                  Product Category    colour     price   price 2  
month                     0.017557 -0.035279  0.012700 -0.014658  
day                      -0.001251  0.005313 -0.002818  0.004254  
order                     0.151693  0.046427 -0.064779  0.000772  
country                   0.132817  0.009789 -0.045345 -0.017536  
session ID                0.016123 -0.032034  0.011669 -0.012742  
Product Category          1.000000  0.230577 -0.364791 -0.137944  
colour                    0.230577  1.000000 -0.091159 -0.080147  
price                    -0.364791 -0.091159  1.000000 -0.743633  
price 2                  -0.137944 -0.080147 -0.743633  1.000000

Strongest correlation we see are the following:

session ID & Month : 0.970961

color and category : 0.230577 

session Id & day : 0.161369

order and category : 0.151693

country and category : 0.132817


In [28]:
#Identify the Data Types
df_info = pd.DataFrame(dataframe.dtypes,columns=['Dtype'])

#Identify the unique values
df_info['Nunique'] = dataframe.nunique()

#check missing values for each column
df_info['MissingValues']=dataframe.isnull().sum()

# Identify the count for each variable
df_info['Count']=dataframe.count()

# Identify the zero values for each variable
df_info['ZeroValues']=(dataframe==0).sum()

# how many '?' values
df_info['?']=(dataframe=='?').sum()

df_info

Dtype  Nunique  MissingValues   Count  ZeroValues  ?
month             int64        5              0  165474           0  0
day               int64       31              0  165474           0  0
order             int64      195              0  165474           0  0
country           int64       47              0  165474           0  0
session ID        int64    24026              0  165474           0  0
Product Category  int64        4              0  165474           0  0
colour            int64       14              0  165474           0  0
price             int64       20              0  165474           0  0
price 2           int64        2              0  165474           0  0